synapseclient/models/agent.py

```diff
@@ -0,0 +1,447 @@
+        self.started_by = synapse_agent_session.get("startedBy", None)
+        self.modified_on = synapse_agent_session.get("modifiedOn", None)
+        self.agent_registration_id = synapse_agent_session.get(
+            "agentRegistrationId", None
+        )
+        self.etag = synapse_agent_session.get("etag", None)
+        return self
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Start_Session: {self.id}"
+    )
+    async def start_async(
+        self, *, synapse_client: Optional[Synapse] = None
+    ) -> "AgentSession":
+        """Starts an agent session.
+
+        Arguments:
+            synapse_client: The Synapse client to use for the request. If None, the default client will be used.
+
+        Returns:
+            The new AgentSession object.
+        """
+        syn = Synapse.get_client(synapse_client=synapse_client)
+        session_response = await start_session(
+            access_level=self.access_level.value,
+            agent_registration_id=self.agent_registration_id,
+            synapse_client=syn,
+        )
+        return self.fill_from_dict(session_response)
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Get_Session: {self.id}"
+    )
+    async def get_async(
+        self, *, synapse_client: Optional[Synapse] = None
+    ) -> "AgentSession":
+        """Gets an agent session.
+
+        Arguments:
+            synapse_client: The Synapse client to use for the request. If None, the default client will be used.
+
+        Returns:
+            The retrieved AgentSession object.
+        """
+        syn = Synapse.get_client(synapse_client=synapse_client)
+        session_response = await get_session(
+            id=self.id,
+            synapse_client=syn,
+        )
+        return self.fill_from_dict(synapse_agent_session=session_response)
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Update_Session: {self.id}"
+    )
+    async def update_async(
+        self,
+        *,
+        access_level: AgentSessionAccessLevel,
+        synapse_client: Optional[Synapse] = None,
+    ) -> "AgentSession":
+        """Updates an agent session. Only updates to the access level are currently supported.
+
+        Arguments:
+            synapse_client: The Synapse client to use for the request. If None, the default client will be used.
+
+        Returns:
+            The updated AgentSession object.
+        """
+        syn = Synapse.get_client(synapse_client=synapse_client)
+
+        self.access_level = access_level
+        session_response = await update_session(
+            id=self.id,
+            access_level=self.access_level.value,
+            synapse_client=syn,
+        )
+        return self.fill_from_dict(session_response)
+
+    @otel_trace_method(method_to_trace_name=lambda self, **kwargs: f"Prompt: {self.id}")
+    async def prompt_async(
+        self,
+        *,
+        prompt: str,
```

Comment:
`*` forces things to be kwargs. It is acceptable that you may want that, but i wanted to call it out if that was not the intention.

I always want synapse_client to be a kwarg so it will never end up after a positional argument